### ISSUE TITLE
ESP32: ws2812 module

### ISSUE THREAD

**Issue by devsaurus** [nodemcu/nodemcu-firmware#1896]
**Rationale:**
Everybody likes colorful LEDs 😉

**Dependencies:**
Again, the RMT peripheral seems to be the best fit for generating teh WS2812 single-wire protocol. Having 8 hardware RMT channels opens up the opportunity to drive up to 8 stripes in parallel at arbitrary pins.

---
API based on ESP8266, changes:
- `ws2812.init()` not needed
- extend `ws2812.write()` to 8 stripes

## ws2812.write({pin = pin1, data = data1}, ...)
Send data via the related pins.

#### Example
```lua
ws2812.write({pin = 4, data = string.char(128,0,0):rep(10)},
             {pin = 16, data = buf1},
             {pin = 5, data = buf2})
```

## ws2812.newBuffer()
Buffer objects unchanged.

**Comment by devsaurus** [comment]
There's a draft implementation in my repo at https://github.com/devsaurus/nodemcu-firmware/tree/dev-esp32-ws2812.

It suffers from the same issues like the `dht` module at the moment:
- espressif/esp-idf#486
- espressif/esp-idf#492

Waiting for the outcome of them.

**Comment by devsaurus** [comment]
Module merged into `dev-esp32` branch.
